src/types-and-traits.rst

```diff
@@ -2735,17 +2787,76 @@ The :t:`lifetime` of an :t:`associated trait constant` shall not be :t:`elided`.
 
 .. rubric:: Examples
 
-:dp:`fls_qtjc7334wzhj`
-Given function ``f`` of the form
+:dp:`fls_2GKCEI9MwMn9`
+Given :t:`static` ``S`` of the form
 
 .. code-block:: rust
 
-   fn f <'a, 'b, T: ToCStr>(&'a mut self, args: &'b [T]) -> &'a mut Command;
+   static S: &[&usize] = &[];
 
-:dp:`fls_vcmmkp9uruhr`
-its lifetime elided form is
+:dp:`fls_f3yZ31dRuTPG`
+its :t:`lifetime` :t:`elided` form is
 
 .. code-block:: rust
 
-   fn f <T: ToCStr>(&mut self, args: &[T]) -> &mut Command;
+   static S: &'static [&'static usize] = &[];
+
+.. _fls_XTBOhK2Yk4lA:
+
+Trait Object Lifetime Elision
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. rubric:: Legality Rules
+
+:dp:`fls_fuBYWRrgxlbQ`
+:t:`Trait object lifetime elision` is a form of :t:`lifetime elision` that
+applies to :t:`[trait object type]s`.
+
+:dp:`fls_URl9CeIVsiWs`
+An :t:`elided` :t:`lifetime` of a :t:`trait object type` is inferred as follows:
+
+* :dp:`fls_SHhw6lYHeYyQ`
+  If the :t:`trait object type` is used as the :t:`type specification` of a
+  :t:`reference type`, then the :t:`lifetime` of the :t:`reference type` is the
+  :t:`elided` :t:`lifetime`,
+
+* :dp:`fls_lC2rwdPLRwaf`
+  If the :t:`trait object type` is used as a :t:`generic argument` and
+
+  * :dp:`fls_e36Hh4oJvfhv`
+    if the corresponding :t:`generic parameter` has exactly one :t:`lifetime
+    bound`, then the :t:`lifetime` of that :t:`bound` is the :t:`elided`
+    :t:`lifetime`,
+
+  * :dp:`fls_ptejalcnIQtm`
+    Otherwise it is a static error to infer the :t:`lifetime` :t:`bound`.
+
+* :dp:`fls_rGbdKtTijby4`
+  If the :t:`trait` of the :t:`trait object type` has exactly one :t:`lifetime
+  bound` specified, then the :t:`lifetime` of that :t:`bound` is the inferred
+  :t:`lifetime`,
+
+* :dp:`fls_JhmQpUoExiNZ`
+  If the :t:`trait` of the :t:`trait object type` has no :t:`[lifetime bound]s`
+  specified, then the :t:`elided` :t:`lifetime` is the ``'static``
+  :t:`lifetime` unless it is :t:`elided` in :t:`[expression]s` where it is
+  instead inferred,
+
+* :dp:`fls_cglZigwAnASl`
+  Otherwise it is a stic error to infer the :t:`lifetime bound`.
+
+.. rubric:: Examples
+
+:dp:`fls_MipY2emZFF6d`
+Given :t:`static` ``S`` of the form
+
+.. code-block:: rust
+
+   type T<'a> = &'a dyn Trait;
+
+:dp:`fls_YPesUZqYHVUX`
+its :t:`lifetime` :t:`elided` form is
```

Comment:
This example is also backwards.

You may or may not want to introduce the vocabulary of "elaborated form" for the opposite of elided form.